test/integration/acceptance/query_permission_test_signatories.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "query_permission_test_signatories.hpp"
+
+QueryPermissionSignatories::QueryPermissionSignatories()
+    : QueryPermissionTestBase({Role::kGetMySignatories},
+                              {Role::kGetDomainSignatories},
+                              {Role::kGetAllSignatories}) {
+  for (int i = 0; i < 5; ++i) {
+    user_signatories_.emplace_back(
+        crypto::DefaultCryptoAlgorithmType::generateKeypair().publicKey());
+  }
+}
+
+IntegrationTestFramework &QueryPermissionSignatories::prepareState(
+    AcceptanceFixture *fixture,
+    const interface::RolePermissionSet &spectator_permissions /*= {}*/) {
+  auto target_permissions = spectator_permissions;
+  target_permissions.set(Role::kAddSignatory);
+
+  // initial state setup
+  auto &itf = QueryPermissionTestBase::prepareState(
+      fixture, spectator_permissions, target_permissions);
+
+  // Add assets to target user
+  for(const auto &public_key : user_signatories_) {
+    itf.sendTxAwait(fixture->complete(fixture->baseTx(kUserId).addSignatory(
+                                          kUserId, public_key),
+                                      kUserKeypair),
+                    getBlockTransactionsAmountChecker(1));
+  }
+
+  return itf;
+}
+
+std::function<void(const proto::QueryResponse &response)>
+QueryPermissionSignatories::getGeneralResponseChecker() {
+  return [this](const proto::QueryResponse &response) {
+    ASSERT_NO_THROW({
+      const auto &resp = boost::apply_visitor(
+          framework::SpecifiedVisitor<interface::SignatoriesResponse>(),
+          response.get());
+
+      const auto &resp_keys = resp.keys();
+      ASSERT_EQ(
+          boost::size(resp_keys),
+          user_signatories_.size() + 1);  // 1 more for user's initial keypair
+
+      /// check that every initially added signatory key is in the result
```

Comment:
afaik triple comments are usually doc-comments

Reply:
yes. I thought I could document the body, but after your comment I checked that I cannot. thanks.